system/src/system_network_diagnostics.cpp

```diff
@@ -80,9 +80,28 @@ class SignalCache
 #endif
     }
 
+#if HAL_PLATFORM_CELLULAR
```

Comment:
I’m not entirely sure this will catch Gen 3 devices. Check the cellular no fixture tests.

Reply:
`HAL_PLATFORM_CELLULAR` is working below to send the messages for Gen3 devices.